src/db/mongodb/mod.rs

```diff
@@ -51,11 +53,28 @@ impl MongoDb {
         })
     }
 
+    /// Creates a collection if it does not exist.
+    pub async fn create_collection<T: MongoDbCollection>(&self) {
+        self.db.create_collection(T::NAME, None).await.ok();
+    }
+
     /// Gets a collection of the provided type.
     pub fn collection<T: MongoDbCollection>(&self) -> T {
         T::instantiate(self, self.db.collection(T::NAME))
     }
 
+    /// Gets all index names by their collection.
+    pub async fn get_index_names(&self) -> Result<HashMap<String, HashSet<String>>, Error> {
+        let mut res = HashMap::new();
+        for collection in self.db.list_collection_names(None).await? {
+            let indexes = self.db.collection::<Document>(&collection).list_index_names().await?;
+            if !indexes.is_empty() {
+                res.insert(collection, indexes.into_iter().collect());
```

Comment:
Couldn't you just move the whole `Vec` into that `HashMap` instead of collecting it into a `HashSet`?

Reply:
ah...I see you want to keep the lookup performance.

Reply:
Well, yes, but part of the check uses `HashSet::difference` so either way I need that.